### PR TITLE
Proposing Krustlet for Sandbox

### PR DESCRIPTION
**Pull request by thomastaylor312**
Placeholder PR for review by other maintainers

### CODE REVIEW

File: proposals/sandbox/krustlet.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+The primary goal of Krustlet is to be the easiest way to integrate Wasm support as seamlessly as
+possible into Kubernetes. It will allow for users to run heterogeneous clusters where both
+containers and Wasm modules can run (though not necessarily on the same node)
+
+## Current Status
+
+Krustlet is very close to a 1.0 release. It was created in late 2019 and has grown along with the
+rest of the Wasm community. Essentially 1.0 is stating that that basic features (but not all of
+them) for a kubelet are in place and the software is reliable enough to use in production. Continued
```
Comment (kflansburg) [inline proposals/sandbox/krustlet.md:21]:
```suggestion
them) for a Kubelet are in place and the software is reliable enough to use in production. Continued
```

**Review by kflansburg — APPROVED** [review]
LGTM

File: proposals/sandbox/krustlet.md (hunk 1 above)
Comment (technosophos) [inline proposals/sandbox/krustlet.md]: Minor clarification needed here: "but not all of..." what? All basic features, all features?

**Review by technosophos — APPROVED** [review]
LGTM. I suggested one clarification, but that is a non-blocker

File: proposals/sandbox/krustlet.md (hunk 1 above)
Comment (thomastaylor312) [reply proposals/sandbox/krustlet.md]: Ok I think I clarified it

File: proposals/sandbox/krustlet.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Krustlet
+
+## Description
+
+Krustlet is a Kubernetes Kubelet implementation for running WebAssembly (Wasm) modules on nodes.
+Krustlet acts as a Kubelet by listening on the event stream for new pods that the scheduler assigns
+to it based on specific Kubernetes tolerations. It also can be extended to enable other types of
+non-Wasm runtimes using a "Provider Model" similar to [Virtual
+Kubelet](https://github.com/virtual-kubelet/virtual-kubelet), but its primary use case is for Wasm.
```
Comment (squillace) [inline proposals/sandbox/krustlet.md]: "but its originating use case"

File: proposals/sandbox/krustlet.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+rest of the Wasm community. Essentially 1.0 is stating that that basic features (but not all
+advanced features such as pod resources, full networking stack integration, health checks) for a
+kubelet are in place and the software is reliable enough to use in production. Continued growth of
+the project will include implementing more of the advanced kubelet features and tracking upstream
+changes to the the WebAssembly Systems Interface spec (WASI).
+
+## Value for the CNCF ecosystem
+
+Krustlet is a great fit for the CNCF ecosystem. First and foremost, it is a Kubernetes-related
+project. That in and of itself is not enough to justify inclusion, but Krustlet has proven to be an
```
Comment (squillace) [inline proposals/sandbox/krustlet.md:30]: it was a part of the recent KubeCon EU Cloud Native WASM day.....

File: proposals/sandbox/krustlet.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+interesting alternative runtime for Kubernetes and has driven interest in the wider Kubernetes
+community around using Wasm. As Sandbox projects are meant to "[e]ncourage public visibility of
```
Comment (squillace) [inline proposals/sandbox/krustlet.md]: wasm and other non-container workloads?

File: proposals/sandbox/krustlet.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+experiments or other early work that can add value to the CNCF mission and build the ingredients of
+a successful Incubation level project," Krustlet is a natural fit as Wasm is an emerging technology
+that many in the CNCF have expressed an interest in. As seen at the latest KubeCon EU, community
+interest in Wasm grew to the point where it merited its own adjacent event. Given the rapidly
```
Comment (squillace) [inline proposals/sandbox/krustlet.md]: right, ok, ignore above comment....

**Review by squillace — APPROVED** [review]
OK, sure seems good to me. Knock yourself out. I think it's mostly straightforward.

File: proposals/sandbox/krustlet.md (hunk 4 above)
Comment (thomastaylor312) [reply proposals/sandbox/krustlet.md]: Yeah, that would probably be good to say as well

**Comment by thomastaylor312** [comment]
Closing PR and opening against main repo

**Pull request closed without merging**